### PR TITLE
KMeans improvements (CQL, hull attr, CLI opts)

### PR DESCRIPTION
**Pull request by blastarr**

### CODE REVIEW

File: analytics/javaspark/src/main/java/mil/nga/giat/geowave/analytic/javaspark/kmeans/KMeansRunner.java [hunk 1]
```diff
@@ -102,14 +107,50 @@ public void run()
 		queryOptions.setAdapter(featureAdapterIds);
 
 		// This is required due to some funkiness in GeoWaveInputFormat
-		queryOptions.getAdaptersArray(inputDataStore.createAdapterStore());
+		AdapterStore adapterStore = inputDataStore.createAdapterStore();
+		queryOptions.getAdaptersArray(adapterStore);
```
Comment (rfecher) [inline analytics/javaspark/src/main/java/mil/nga/giat/geowave/analytic/javaspark/kmeans/KMeansRunner.java]: really? thats not good if its true

File: analytics/javaspark/src/main/java/mil/nga/giat/geowave/analytic/javaspark/kmeans/KMeansRunner.java [hunk 2]
```diff
@@ -102,14 +107,50 @@ public void run()
...
+				}
+
+				if ((bbox != null) && !bbox.equals(GeometryUtils.infinity())) {
+					query = new SpatialQuery(
+							bbox);
+				}
+			}
+		}
+		catch (CQLException e) {
+			LOGGER.error("Unable to parse CQL: " + cqlFilter);
```
Comment (rfecher) [inline analytics/javaspark/src/main/java/mil/nga/giat/geowave/analytic/javaspark/kmeans/KMeansRunner.java]: ideally we'd have a stack trace (ie. add `, e);`  Not worth disapproving the PR jsut for that.

File: analytics/javaspark/src/main/java/mil/nga/giat/geowave/analytic/javaspark/kmeans/KMeansRunner.java [hunk 3]
```diff
@@ -102,14 +107,50 @@ public void run()
...
+				}
+
+				DataAdapter adapter = adapterStore.getAdapter(cqlAdapterId);
+
+				if (adapter instanceof FeatureDataAdapter) {
+					final String geometryAttribute = ((FeatureDataAdapter) adapter)
+							.getFeatureType()
+							.getGeometryDescriptor()
+							.getLocalName();
+					Filter filter;
```
Comment (rfecher) [inline analytics/javaspark/src/main/java/mil/nga/giat/geowave/analytic/javaspark/kmeans/KMeansRunner.java]: odd, must have been an artifact of something OBE for these two lines to be separate. iNot worth disapproving the PR just for that.

File: analytics/javaspark/src/main/java/mil/nga/giat/geowave/analytic/javaspark/kmeans/KMeansRunner.java (hunk 1 above)
Comment (blastarr) [reply analytics/javaspark/src/main/java/mil/nga/giat/geowave/analytic/javaspark/kmeans/KMeansRunner.java]: That's an old comment. I have no idea why it's there. I think I split it out just to get a ref to the adapterStore for reuse farther down.

**Pull request merged**